tests/pmap_test.py

```diff
@@ -610,6 +610,23 @@ def testGatherReplicaGroupsInterleaved(self):
 
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  def testGradOfGather(self):
```

Comment:
Would it be possible to use the `jtu.check_grads` function, perhaps patterning off of `testGradOfPsum`? That way we'll get slightly more general testing (e.g. we don't need the `jnp.sum` in the definition of `g`, and so we'll be testing the transpose rule with more than just all-equal cotangent vector entries.)

Reply:
Done. Thanks for the tip.